app/screens/home/channel_list/servers/servers_list/server_item/index.tsx

```diff
@@ -0,0 +1,242 @@
+import DatabaseManager from '@database/manager';
+import WebsocketManager from '@init/websocket_manager';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+import {typography} from '@utils/typography';
+import {removeProtocol, stripTrailingSlashes} from '@utils/url';
+
+import Options from './options';
+
+import type ServersModel from '@typings/database/models/app/servers';
+import type MyChannelModel from '@typings/database/models/servers/my_channel';
+import type {Subscription} from 'rxjs';
+
+type Props = {
+    isActive: boolean;
+    server: ServersModel;
+}
+
+type BadgeValues = {
+    isUnread: boolean;
+    mentions: number;
+}
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => ({
+    active: {
+        borderColor: theme.sidebarTextActiveBorder,
+        borderWidth: 3,
+    },
+    badge: {
+        left: 18,
+        top: -5,
+    },
+    button: {
+        borderRadius: 8,
+        flex: 1,
+        height: 72,
+        justifyContent: 'center',
+        paddingHorizontal: 18,
+    },
+    container: {
+        alignItems: 'center',
+        backgroundColor: changeOpacity(theme.centerChannelColor, 0.04),
+        borderRadius: 8,
+        flexDirection: 'row',
+        height: 72,
+        marginBottom: 12,
+    },
+    details: {
+        marginLeft: 14,
+    },
+    logout: {
+        backgroundColor: theme.centerChannelBg,
+        borderRadius: 8,
+        height: 16,
+        left: 40,
+        position: 'absolute',
+        top: 11,
+        width: 16,
+    },
+    nameContainer: {
+        alignItems: 'center',
+        flexDirection: 'row',
+    },
+    name: {
+        color: theme.centerChannelColor,
+        ...typography('Body', 200, 'SemiBold'),
+    },
+    offline: {
+        opacity: 0.5,
+    },
+    row: {flexDirection: 'row'},
+    unread: {
+        top: -2,
+        left: 25,
+    },
+    url: {
+        color: changeOpacity(theme.centerChannelColor, 0.72),
+        ...typography('Body', 75, 'Regular'),
+    },
+    websocket: {
+        marginLeft: 7,
+        marginTop: 4,
+    },
+}));
+
+const ServerItem = ({isActive, server}: Props) => {
+    const intl = useIntl();
+    const theme = useTheme();
+    const [badge, setBadge] = useState<BadgeValues>({isUnread: false, mentions: 0});
+    const styles = getStyleSheet(theme);
+    const swipeable = useRef<Swipeable>();
+    const activeStyle = isActive ? styles.active : undefined;
+    const offlineStyle = server.lastActiveAt ? undefined : styles.offline;
+    const websocketError = server.lastActiveAt ? !WebsocketManager.isConnected(server.url) : false;
+    let displayName = server.displayName;
+    if (server.url === server.displayName) {
+        displayName = intl.formatMessage({id: 'servers.default', defaultMessage: 'Default Server'});
+    }
+
+    const unreadsSubscription = (myChannels: MyChannelModel[]) => {
+        let mentions = 0;
+        let isUnread = false;
+        myChannels.forEach((myChannel) => {
```

Comment:
Probably the same, but this could have been solved by a `reduce`, instead of a `forEach`.

Reply:
100% true, but in JS context `forEach` is faster than `reduce` in fact I may even change this to use `for .. of ...` as is even faster.